tools/debian/convert_debian.py

```diff
@@ -51,227 +56,307 @@
-        val_data = file_path_map.get(mapped_key_no_ext + '.data')
-
-        if not val_wml:
-            # TODO: Fill out partial advisory schema
-            print('No WML file yet for this: ' + mapped_key_no_ext)
-            continue
+def write_output(output_dir: str, advisories: Advisories):
+  """Writes the advisory dict into individual json files"""
+  for dsa_id, advisory in advisories.items():
+    if advisory.preexisting:
+      continue
 
-        with open(val_wml, encoding='utf-8') as handle:
-            data = handle.read()
-            html = WML_DESCRIPTION_PATTERN.findall(data)[0]
-            res = markdownify.markdownify(html)
-            advisory.details = res
+    with open(
+        os.path.join(output_dir, dsa_id + '.json'), 'w',
+        encoding='utf-8') as output_file:
+      output_file.write(json.dumps(advisory, default=dumper, indent=2))
+      print(
+          'Writing: ' + os.path.join(output_dir, dsa_id + '.json'), flush=True)
 
-        with open(val_data, encoding='utf-8') as handle:
-            data: str = handle.read()
-            report_date: str = WML_REPORT_DATE_PATTERN.findall(data)[0]
+  print('Complete')
 
-            # TODO: Handle multiple dates? What does the
-            #  multiple report dates mean
-            # if len(report_date.split(',')) > 1:
-            #     print(report_date)
-            #     print(dsa_id)
 
-            # Split by ',' here for the occasional case where there
-            # are two dates in the 'publish' field
-            advisory.published = (datetime.datetime.strptime(
-                report_date.split(',')[0], '%Y-%m-%d').isoformat() + 'Z')
+def is_dsa_file(name: str):
+  """Check if filename is a DSA output file, e.g. DSA-1234-1.json"""
+  return name.startswith('DSA-') and name.endswith('.json')
 
 
-def write_output(output_dir: str, advisories: Advisories):
-    """Writes the advisory dict into individual json files"""
-    for dsa_id, advisory in advisories.items():
-        with open(os.path.join(output_dir, dsa_id + '.json'),
-                  'w',
-                  encoding='utf-8') as output_file:
-            output_file.write(json.dumps(advisory, default=dumper, indent=2))
-            print('Writing: ' + os.path.join(output_dir, dsa_id + '.json'),
-                  flush=True)
+def load_advisories(json_dir: str, advisories: Advisories):
+  """Loads the existing converted advisories"""
+  for file in filter(is_dsa_file, os.listdir(json_dir)):
+    with open(os.path.join(json_dir, file), encoding='utf-8') as handle:
+      # SimpleNamespace loads in the json as a
+      # python object instead of a dict
+      loaded_advisory: AdvisoryInfo = json.loads(
+          handle.read(), object_hook=lambda d: SimpleNamespace(**d))
+      if not hasattr(loaded_advisory,
+                     'modified') or not loaded_advisory.modified:
+        continue
 
-    print('Complete')
+      loaded_advisory.preexisting = True
```

Comment:
nit: this adds a fair bit of complexity, and is not really necessary is it? we can remove it if so. 

I expect we'll be running this from scratch most of the time.